### ISSUE TITLE
I want to use multi-node training, but the process is stuck

### ISSUE THREAD

**Issue by pixeli99** [huggingface/diffusers#4430]
![image](https://github.com/huggingface/accelerate/assets/46072190/34e86cc0-ca7e-4c42-a4d5-dbc814bfe666)
![image](https://github.com/huggingface/accelerate/assets/46072190/fdb99324-f1a4-4e47-96f8-2fe65b9f12ff)

The thing is, I use multiple machines, 2x6 A100, to train controlnet, but I don't quite understand why the process gets stuck where I marked the red box and can't move on.

When I checked what cpu was doing when I was stuck, I found that ```output clock_gettime()```(using `strace -p 33104`)

I'm not sure how to solve this problem, which bothers me for a long time. I guess it's main process first's problem, so I got rid of it, which does keep the program running, but I don't understand the principle, and he will continue to get stuck as the following picture shows.
![image](https://github.com/huggingface/accelerate/assets/46072190/3076a5cf-ad49-4810-8ad9-9faff26c44f2)
code here https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet_sdxl.py

In other words, if I use IB, the process will block, there will be no problem with tcp, and I should be able to make sure that my ib communication is fine.

**Comment by sayakpaul** [comment]
Cc: @lhoestq. 

https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet_sdxl.py#L1001C1-L1007C112

`compute_embeddings_fn()` in the block above uses GPUs. Is this a problem when running the script with multiple GPUs?

**Comment by lhoestq** [comment]
I don't think so but I'm not familiar with accelerate

**Comment by muellerzr** [comment]
Duplicate issue opened on accelerate here: https://github.com/huggingface/accelerate/issues/1804 I'll be looking into it
